src/main/kotlin/org/rust/lang/refactoring/implementMembers/impl.kt

```diff
@@ -80,28 +79,26 @@ private fun insertNewTraitMembers(selected: Collection<RsAbstractable>, existing
         }
 
         val anchor = indexedExistingMembers
-            .map { IndexedValue(it.index, it.value.first) }
-            .lastOrNull()
+            .map {
+                val member = it.value.first
+                IndexedValue(it.index, member.expandedFromRecursively ?: member)
+            }.lastOrNull()
             ?: IndexedValue(-1, existingMembers.lbrace) ?: return
 
         val addedMember = existingMembers.addAfter(newMember, anchor.value) as RsAbstractable
         existingMembersWithPosInTrait.add(anchor.index + 1, Pair(addedMember, posInTrait))
 
         // If the newly added item is a function, we add an extra line between it and each of its siblings.
-        val previousAbstractable = addedMember.leftSiblings.find { it is RsAbstractable }
-        if (previousAbstractable != null) {
-            if (previousAbstractable is RsFunction || addedMember is RsFunction) {
-                val whitespaces = createExtraWhitespacesAroundFunction(previousAbstractable, addedMember)
-                existingMembers.addBefore(whitespaces, addedMember)
-            }
+        val prev = addedMember.leftSiblings.find { it is RsAbstractable || it is RsMacroCall }
```

Comment:
Those two pieces are almost identical, maybe we should extract a function here

Reply:
There are a lot of differences: leftSiblings/rightSiblings prev/nex/addedMember order in `createExtraWhitespacesAroundFunction` call parameters, addBefore/addAfter. So I don't know how to generalize it.

Reply:
You're right, missed that